sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,192 @@
+        elif data.ndim is 1:
+            result = Cube1D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        else:
+            result = NDCube(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        return result
+
+
+class Cube2D(NDCube):
+    """docstring for Cube2D"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        super(Cube2D, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def plot(self, axes=None, axis_data=['x', 'y'], **kwargs):
+        """
+        Plots an x-y graph at a certain specified wavelength onto the current
+        axes. Keyword arguments are passed on to matplotlib.
+
+        Parameters
+        ----------
+        axes: `astropy.visualization.wcsaxes.core.WCSAxes` or `None`:
+            The axes to plot onto. If None the current axes will be used.
+
+        axis_data: `list`.
+            The first axis in WCS object will become the first axis of axis_data and
+            second axis in WCS object will become the seconf axis of axis_data.
+        """
+        if axes is None:
+            if self.wcs.naxis is not 2:
+                slice_list = self.wcs._naxis
+                axis_index = []
+                for i, ax in enumerate(slice_list):
+                    if ax is not 1:
+                        axis_index.append(i)
+                if len(axis_index) is 2:
+                    slice_list[axis_index[0]] = axis_data[0]
+                    slice_list[axis_index[1]] = axis_data[1]
+                else:
+                    raise ValueError("Dimensions of WCS and data don't match")
+            axes = wcsaxes_compat.gca_wcs(self.wcs, slices=slice_list)
+        plot = axes.imshow(self.data, **kwargs)
+        return plot
+
+
+class Cube1D(NDCube):
+    """docstring for Cube1D"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        super(Cube1D, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def plot(self, unit=None, origin=0):
+        """
+        Plots a graph.
+        Keyword arguments are passed on to matplotlib.
+
+        Parameters
+        ----------
+        unit: `astropy.unit.Unit`
+        The data is changed to the unit given or the self.unit if not given.
+        """
+        index_not_one = []
+        for i, item in enumerate(self.wcs._naxis):
+            if item is not 1:
+                index_not_one.append(i)
+        if unit is None:
+            unit = self.wcs.wcs.cunit[index_not_one[0]]
+        plot = plt.plot(self.pixel_to_world(origin=origin) * unit)
```

Comment:
Replace this line with
```plot = plt.plot(self.pixel_to_world(Quantity(np.arange(len(self.data)), unit=u.pix), origin=origin).to(unit), self.data)```
where ```self.pixel_to_world``` needs to take the pixel values as an input.  See comment on ```self.pixel_to_world```